File: application-web-mvc/src/main/java/com/osori/mvc/controller/AccountController.java
package com.osori.mvc.controller;

import com.osori.mvc.domain.Account;
import com.osori.mvc.repository.AccountRepository;
import com.osori.mvc.service.AccountService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class AccountController {
    private final AccountService accountService;

    @GetMapping("/account/{username}/{password}/{role}")
    public Account insert(@ModelAttribute Account account) {
        return accountService.createNew(account);
    }
}

File: application-web-mvc/src/main/kotlin/com/osori/mvc/controller/AccountController.kt
package com.osori.mvc.controller

import com.osori.mvc.domain.Account
import com.osori.mvc.service.AccountService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.RestController

@RestController
class AccountController (private val accountService: AccountService) {


    @GetMapping("/account/{username}/{password}/{role}")
    fun insert(@ModelAttribute account: Account): Account {
        return accountService.createNew(account)
    }
}